lib/App/git/ship/perl.pm

```diff
@@ -139,7 +139,7 @@ sub ship {
   $self->system(qw( git add Makefile.PL ), $changelog, $self->_filename('readme'));
   $self->system(qw( git commit -a -m ), $self->_changes_to_commit_message);
   $self->SUPER::ship(@_);    # after all the changes
-  $uploader->upload_file($dist_file);
+  $uploader->upload_file($dist_file) unless $ENV{GIT_SHIP_NO_UPLOAD};
```

Comment:
turn $uploader into an attribute
- :+1:

unless $ENV{GIT_SHIP_UPLOAD} // 1;
- `do the upload unless UPLOAD is set`, right?  So if I set UPLOAD, it won't upload?  If my brain logic is functioning properly, that seems counter-intuitive.  I was thinking `do the upload unless NO_UPLOAD is set`, or, said another way, `if NO_UPLOAD is set, don't upload`.